### PR TITLE
fix: proxy policies middleware tus filename evaluation

### PR DESCRIPTION
**Pull request by fschade**
## Description
when uploading files via uppy (tus), the path does not give any information about the file, PUT contains the filename in the path, tus POST not.

this pr extracts the HeaderUploadMetadata from that POST request and enhances the policies grpc environment request with that information. 

Therefore, the policies service is now able to evaluate proxy requests for tus uploads too. 

## Open tasks

* [ ] add unit tests
* [ ] confirm that the implementation is performant

### CODE REVIEW

**Comment by fschade** [comment]
@micbar, i had not much time... therefore no unit tests at the moment, no changelog neither... i can add those tomorrow, can you please check within your installation and confirm it works!? 🍺

**Review by micbar — APPROVED** [review]
LGTM

File: deployments/examples/service_policies/policies/proxy.rego [hunk 1]
```diff
@@ -3,10 +3,16 @@ package proxy
...
-granted := false if {
+granted = false if {
     utils.is_request_type_put
     not input.request.path == "/data"
     not utils.collection_contains(utils.ALLOWED_FILE_EXTENSIONS, input.request.path)
 }
+
+granted = false if {
+    utils.is_request_type_post
+    startswith(input.request.path, "/remote.php")
```
Comment (kobergj) [inline deployments/examples/service_policies/policies/proxy.rego:16]: This seems wrong. I can (and should) omit the `remote.php` part when I send requests to ocis. What is the point of it?

**Review by kobergj — CHANGES_REQUESTED** [review]

File: deployments/examples/service_policies/policies/proxy.rego (hunk 1 above)
Comment (micbar) [reply deployments/examples/service_policies/policies/proxy.rego:16]: There is another POST request for the login  which should not be blocked by this rule.

Comment (kobergj) [reply deployments/examples/service_policies/policies/proxy.rego:16]: I don't understand. My point was that checking for `remote.php` doesn't bring any value. Either requests of a client will start with `remote.php` or none will. It is even possible that both is true. So I don't understand why we check for it in the policies?

Comment (kobergj) [reply deployments/examples/service_policies/policies/proxy.rego:16]: Ja ok fine. Just the example makes no sense. But it is not critical. We can go with it.

**Review by kobergj — APPROVED** [review]

File: deployments/examples/service_policies/policies/proxy.rego (hunk 1 above)
Comment (fschade) [reply deployments/examples/service_policies/policies/proxy.rego:16]: `POST - https://localhost:9200/konnect/v1/token`
`POST - https://localhost:9200/api/v0/settings/roles-list`
`POST - https://localhost:9200/api/v0/settings/roles-list`

should not be blocked.... use urls to identify the type is not the best solution, i agree.

i can hydrate the environment if it's a upload and or create request. We can iterate on it.

Comment (kobergj) [reply deployments/examples/service_policies/policies/proxy.rego:16]: I was just about the `remote.php` part. This makes no sense as I can send all requests also without it and would still get the same results. But you are right, let's iterate over it when we see fit.

**Pull request merged**